### DISCUSSION TITLE
How to block all NON-English Words and Sentences

### DISCUSSION THREAD

**Discussion by shrijayan** [NVIDIA-NeMo/Guardrails#830] [General]
I want to block input prompts that contain non-English words or characters. How can I do this in Nemo Guardrails?

**Comment by Pouyanpi** [comment]
Hi @shrijayan, there are potentially different ways you can achieve this. If the prompt contains any non English word or character then you want to block that prompt from being forwarded to the LLM?

You can define an input rails where it executes an action that checks wether the input contains non English words/chars. This check can be done via a call to LLM or other simpler functions. To get an idea, check `examples/bots/abc` where it uses `self check input`, you can see the code in `nemoguardrails/library/self_check/input_check`.

If you express what you intend to achieve with an example we can get more detailed, but defining an input rails with a custom action might be a good starting point.
